### DISCUSSION TITLE
Proposed Bridge Element mapping to IFC4.3.x

### DISCUSSION THREAD

**Discussion by jwouellette** [jwouellette/TPF-5_372-Unit_Test_Suite#2] [General]
Please refer to the [spreadsheet ](https://github.com/jwouellette/TPF-5_372-Unit_Test_Suite/blob/main/Proposed%20IFC%20mapping%20for%20bridge%20elements.xlsx) and provide feedback.

**Comment by RickBrice** [comment]
I seem to be preferring Option #2 for the spatial structure breakdown. Approaches, slope protection, embankment fill walls, and other ancillary items are related to the bridge and not of the bridge themselves. However, I think there should be an IfcRelationship that ties the ancillary items to the bridge.

1. Should "foundations" be modelled concept or ignored? - Should be modelled since this is a design to construction exchange, contractor needs to know details about foundations to accurate bid. This is a big ask, so a more simple "version 1" solution might be more practical
2. Should Deck Joints be modeled? - Same response as question 1
3. Are Bolds modelled or just the Bolt Holes? - Bold location and quantities should be modeled. Bold specifications (size, type, materials, etc) don't need a detailed geometric representation in models
4. Should all aggregations be IfcElementAssembly or just have sub-elements use IfcRelAssociates? - I don't have a deep enough understanding to weigh the pro's and con's. My gut tells me that IfcElementAssembly is easier but IfcRelAssociates is more flexible and can allow sub-elements to be referenced into multiple places in a model reducing data size.

**Reply by jwouellette** [reply]
1. We'll explore an alternative method. The problem is not being able to have two different hierarchies simultaneously. Seems the primary idea is to first have Superstructure-Substructure and then define Abutment and Piers inside the Substructure. Maybe we group the "foundation" elements inside IfcBridgePart.PIER as an IfcElementAssembly? Or, we add another layer of spatial hierarchy under IfcBridgePart.PIER and IfcBridgePart.ABUTMENT as IfcBridgePart.FOUNDATION. 

2. The vendors are saying that by modeling the construction joints at this stage, the parametric "deck" inside the authoring software would have to be "exploded" and thus no longer parametric. The Contractor/Fabrication model could show the joints and deck segments, since this is where sequence and details will be finalized.

3. I think early reactions from other vendors include modeling bolt HOLE locations, but NOT modeling the bolts themselves and just defining them in drawings and specifications. The Fabricator/Detailer would then provide the detailed "shop models" indicating the fasteners in full geometric glory. Is this agreeable?

4. I think this also relates to Question 1 and depends on the consensus of how deep a spatial hierarchy (IfcBridgePart within IfcBridgePart) could/should be... 2 levels? 3 levels? 4 levels?

**Comment by RickBrice** [comment]
Digging into the IFC breakdown for 'Girders (Concrete)', it is proposed that IfcReinforcingBar.LIGATURE be used for precast/prestressed concrete lifting loops. I'm not sure this is the best or most clear representation, or if the representation is even needed other than to indicate the location of lifting devices.

Lifting devices (loops or other) are typically designed by fabricators because they are means and methods of handling the girder. Some DOTs will define the location of lifting loops for handling of the girder, but the details are left to the fabricator. Lifting loops are often unstressed 7-wire strand, sometimes with multiple strands per loop and sometimes placed in a conduit to ensure uniform load distribution amongst the strands.

A ligature is something for binding or connecting two things together. That is not the function of lifting devices. Would a .USERDEFINED. IfcDiscreteAccessory be a more descriptive type?

**Reply by jwouellette** [reply]
Thanks for the feedback, @RickBrice.
Maybe [IfcDiscreteAccessory](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcDiscreteAccessory.htm) would work better? When you read the documentation, it looks like that is the better fit. The other question is whether or not it should be included in the model from the Bridge Engineer/Designer to the Contractor and Fabricator if the Fabricator is ultimately responsible for the design of it.

I'm not sure why "WireLoop" is missing from the IfcDiscreteAccessoryTypeEnum list, but does have a Pset, [Pset_DiscreteAccessoryWireLoop](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/Pset_DiscreteAccessoryWireLoop.htm).

I'll pose the question on the IFC4.3 documentation repo.

**Comment by VasGiannGit** [comment]
Although being flexible in principle to support any option, it makes more sense to us to go with option #2.

**Comment by RickBrice** [comment]
Not sure if IfcDiscreteAccessory and Pset_DiscreteAccessoryWireLoop makes sense for Deck -> Slab -> Reinforcing. If the intent is to describe bars used to lift and handle precast deck and subdeck panels, the reinforcement is typically standard reinforcing bars. Wire lifting loops are not typically used in precast deck/subdeck panels.

Here is a link to TxDOT subdeck panels https://ftp.dot.state.tx.us/pub/txdot-info/cmd/cserve/standard/bridge/MS-PCP-FAB-19.pdf
The lifting loops are standard #3 bars.
